tests/actions/SessionTest.js

```diff
@@ -0,0 +1,74 @@
+import {fetchAccountDetails, signIn} from '../../src/libs/actions/Session';
+import * as API from '../../src/libs/API';
+import HttpUtils from '../../src/libs/HttpUtils';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+
+// Set up manual mocks for methods used in the actions so our test does not fail.
+jest.mock('../../src/libs/Notification/PushNotification', () => ({
+    register: () => {},
+}));
+
+// We are mocking this method so that we can later test to see if it was called and what arguments it was called with.
+// We test HttpUtils.xhr() since this means that our API command turned into a network request and isn't only queued.
+jest.mock('../../src/libs/HttpUtils', () => ({
+    xhr: jest.fn(),
+}));
+
+test('Authenticate is called with saved credentials when a session expires', () => {
+    const TEST_USER_LOGIN = 'test@testguy.com';
+
+    // Set up mock responses for all APIs that will be called. The next time this command is called it will return
+    // jsonCode: 200 and the response here.
+    API.setMockCommand('GetAccountStatus', jest.fn(() => Promise.resolve({
+        jsonCode: 200,
+        accountExists: true,
+        canAccessExpensifyCash: true,
+        requiresTwoFactorAuth: false,
+    })));
+
+    // Simulate user entering their login and populating the credentials.login
+    fetchAccountDetails(TEST_USER_LOGIN);
+
+    // Note: In order for this test to work we must return a promise! It will pass even with
+    // failing assertions if we remove the return keyword.
+    return waitForPromisesToResolve()
+        .then(() => {
+            // Next we will simulate signing in and make sure all API calls in this flow succeed
+            const authenticateMock = jest.fn(() => Promise.resolve({
```

Comment:
Is there a reason that this uses `jest.fn()` but `CreateLogin` doesn't?

Reply:
Yep! So, another reason to use a `jest.fn()` is that it allows us make assertions against the mock function to see what arguments were passed, how many times it was called, etc. In this case, we aren't interested in testing this behavior, but we still need to provide a mock response so that our credentials will be set in the `signIn()` action.